integration/spark-common/src/main/scala/org/apache/spark/rdd/CarbonMergeFilesRDD.scala

```diff
@@ -157,21 +157,23 @@ object CarbonMergeFilesRDD {
     if (carbonTable.isHivePartitionTable && !StringUtils.isEmpty(tempFolderPath)) {
       // remove all tmp folder of index files
       val startDelete = System.currentTimeMillis()
-      val numThreads = Math.min(Math.max(partitionInfo.size(), 1), 10)
-      val executorService = Executors.newFixedThreadPool(numThreads)
-      val carbonSessionInfo = ThreadLocalSessionInfo.getCarbonSessionInfo
+      // get all files in all folder of index files
+      val allTmpFiles = partitionInfo
+        .asScala
+        .map { partitionPath =>
+          FileFactory.getCarbonFile(partitionPath + "/" + tempFolderPath)
+            .listFiles().toList
+        }.toList.flatten.map(_.getAbsolutePath)
+      // delete files in parallel
+      sparkSession.sparkContext.parallelize(allTmpFiles).map {
```

Comment:
suggest to add a threshold check, use RDD only when `allTmpFiles.length` exceed threshold